core/src/main/java/bio/terra/pearl/core/service/survey/AnswerProcessingService.java

```diff
@@ -56,7 +50,11 @@ public void processAllAnswerMappings(
             return;
         }
         processProfileAnswerMappings(enrollee, answers, mappings, operator, auditInfo);
-        processProxyProfileAnswerMappings(enrollee, answers, mappings, ppUser, auditInfo);
+        // if the ppUser is not the same as the enrollee, we're in a proxy environment,
+        // so we should check proxy profile mappings. otherwise, ignore them.
+        if (!ppUser.getProfileId().equals(enrollee.getProfileId())) {
```

Comment:
Let's move this logic check to inside the processProxy... function, so that function owns proxy related logic